[BUG]: test_http_server intermittent CI failures

### Version

23.11

### Which installation method(s) does this occur on?

_No response_

### Describe the bug.

```
FAILED common/test_http_server.py::test_simple_request[HTTPStatus.OK-text/plain-OK-8-True-True-GET-8088-/test] - AssertionError: Expected 'mock' to be called once. Called 0 times.
4175
Calls: [call.__str__()].
```

### Minimum reproducible example

```shell
Open a PR
```


### Relevant log output

<details><summary>Click here to see error details</summary><pre>

 [Paste the error here, it will be hidden by default]

</pre></details>


### Full env printout

<details><summary>Click here to see environment details</summary><pre>

 [Paste the results of print_env.sh here, it will be hidden by default]

</pre></details>


### Other/Misc.

_No response_

### Code of Conduct

- [X] I agree to follow Morpheus' Code of Conduct
- [X] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

--- a/tests/common/test_http_server.py
+++ b/tests/common/test_http_server.py
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 import os
+import time
 import typing
 from http import HTTPStatus
 from unittest import mock
@@ -83,6 +84,18 @@ def test_simple_request(port: int,
 
         parse_fn.assert_called_once_with(payload)
 
+        if use_callback:
+            # Since the callback is executed asynchronously, we don't know when it will be called.
+            expected_call = mock.call(False, "")
+
+            max_retries = 300
+            attempt = 0
+            while expected_call not in callback_fn.mock_calls and attempt < max_retries:
+                attempt += 1
+                time.sleep(0.1)
+
+            callback_fn.assert_called_once_with(False, "")
+
     if use_context_mgr:
         with HttpServer(parse_fn=parse_fn, port=port, endpoint=endpoint, method=method,
                         num_threads=num_threads) as server:
@@ -99,11 +112,6 @@ def test_simple_request(port: int,
         server.stop()
 
     assert not server.is_running()
-
-    # Since the callback is executed asynchronously, we don't know when it will be called.
-    # However server.stop() will wait for the callback to be executed.
-    if use_callback:
-        callback_fn.assert_called_once_with(False, "")
 
 
 def test_constructor_errors():

